wire-compiler/src/main/java/com/squareup/wire/java/ServiceFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.java;
+
+import com.squareup.javapoet.TypeSpec;
+import com.squareup.wire.model.WireService;
+import java.util.List;
+
+/**
+ * Generates a Java service class for a .proto RPC service. Implementing classes must have a no
+ * arguments constructor.
+ */
+public interface ServiceFactory {
```

Comment:
This is the minimum interface to get us to JavaPoet, but we could probably take it even further. It'd be great to make this thing much more general-purpose, so that plugins could generate whatever they wanted.